ext/openssl/ossl_cipher.c

```diff
@@ -319,7 +319,7 @@ ossl_cipher_pkcs5_keyivgen(int argc, VALUE *argv, VALUE self)
     iter = NIL_P(viter) ? 2048 : NUM2INT(viter);
     if (iter <= 0)
 	rb_raise(rb_eArgError, "iterations must be a positive integer");
-    digest = NIL_P(vdigest) ? EVP_md5() : ossl_evp_get_digestbyname(vdigest);
+    digest = NIL_P(vdigest) ? EVP_sha256() : ossl_evp_get_digestbyname(vdigest);
```

Comment:
This is a breaking change and can't be merged. The way to go would be to require the user to specify the hash function explicitly if EVP_md5() is unavailable.

Reply:
hi, here's the original evp.h
https://gist.github.com/ahorek/9ec56a3faae81a1e547e0de7f0acb7db#file-evp-h-L710

this is the default version in solaris 11 https://www.oracle.com/technetwork/server-storage/solaris11/downloads/index.html

it looks like, it's not deprecated anymore in current versions of openssl

the patch is obviously wrong, I just waned to point to places where the problem is.
I don't have an idea how to detect if these functions are deprecated or not, so I need some help here.
a possible workaround is to remove -Werror=deprecated-declarations flag from the build process, but I don't think it's a good idea :)

Reply:
So, if they're not deprecated, can we close this for now?

When we need such guards, we'll add `OpenSSL.check_func("EVP_md5", "openssl/evp.h")` to ext/openssl/extconf.rb (It's not obvious from the name, though, it checks the existence of the function using `-Werror=deprecated-declarations`.), so that `HAVE_EVP_MD5` macro will be defined only when EVP_md5() exists and is not marked as deprecated.

Reply:
Right now this issue prevents the successful build on solaris with the default openssl lib where these functions ARE marked as deprecated.
thanks for the hint, I'll try that. Closing this for now.